internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -985,6 +976,141 @@ public CompletableFuture<PubSubProduceResult> put(
     return produceResultFuture;
   }
 
+  /**
+   * This is the main method to send DIV messages to a kafka topic through VeniceWriter. The method decides whether to
+   * send the messages in chunked or non-chunked mode based on the size of the message. Today, DIV is the only user of
+   * this method, but it can be extended easily to support other class types in the future.
+   *
+   * All the messages sent through this method are of type {@link MessageType#CONTROL_MESSAGE_DIV} in its KafkaKey and
+   * all their corresponding {@link KafkaMessageEnvelope} uses {@link Put} as the payload. Inside the Put payload, the
+   * actual message is stored in the putValue field and the schema id has 3 cases:
+   *
+   * 1. If the message is non-chunked, the schema id is set to {@link AvroProtocolDefinition#GLOBAL_DIV_STATE}.
+   * 2. If the message is chunk message, the schema id is set to {@link AvroProtocolDefinition#CHUNK}.
+   * 3. If the message is a chunk manifest message, the schema id is set to {@link AvroProtocolDefinition#CHUNKED_VALUE_MANIFEST}.
+   */
+  public CompletableFuture<PubSubProduceResult> sendChunkSupportedDivMessage(int partition, K key, V value) {
+    if (partition < 0 || partition >= numberOfPartitions) {
+      throw new VeniceException("Invalid partition: " + partition);
+    }
+
+    byte[] serializedKey = keySerializer.serialize(topicName, key);
+    byte[] serializedValue = valueSerializer.serialize(topicName, value);
+    int totalRecordSize = calculateTotalRecordSize(serializedKey, serializedValue, null);
+
+    if (isChunkingNeededForRecord(totalRecordSize)) {
+      return sendDivMessageChunked(
+          partition,
+          serializedKey,
+          serializedValue,
+          GLOBAL_DIV_STATE.getCurrentProtocolVersion(),
+          null);
+    } else {
+      return sendDivMessageNonChunked(
+          partition,
+          serializedKey,
+          serializedValue,
+          GLOBAL_DIV_STATE.getCurrentProtocolVersion(),
+          null);
+    }
+  }
+
+  private CompletableFuture<PubSubProduceResult> sendDivMessageChunked(
```

Comment:
I feel like these functions have very large overlap with the existing functions, and the existing ones can be repurposed to achieve the new functionality. But I think I'll just try it out when I'm working on it.
![Screenshot 2024-12-05 at 5 06 27 PM](https://github.com/user-attachments/assets/fb264367-1517-4102-9763-b172bcb46296)